File: cpp/tsubame.cpp
#include <iostream>
using namespace std;
 
int main(){
    // 数字を取得する(文字列として)
    string str;
    getline(cin, str);
    auto num = stoi(str);
     
    // 10〜99以外の場合は例外を投げる
    const auto min = 10;
    const auto max = 99;
    if (num < min || max < num) {
        throw "The number must be " + to_string(min) + " ... " + to_string(max) +".";
    }
     
    // それぞれの桁数を計算する
    // (文字として扱われるため、必ず'0'の文字コードの数値を減算する)
    const auto zero = '0';
    for (auto ch : str) {
        num += ch - zero;
    }
     
    // 数値を出力する
    cout << num << endl;
     
    return 0;
}

File: cs/Tsubame.cs
using System;

public class Tsubame {
    public static void Main(){
        // 文字列を取得し、整数に変換する
        var line = System.Console.ReadLine();
        int num = int.Parse(line);

        // 不正な値なら、例外を投げる
        const int minNum = 10;
        const int maxNum = 99;
        if (num < minNum || maxNum < num) {
            throw new Exception("The number must be " + minNum + "..." + maxNum + ".");
        }

        // それぞれの桁の数値を加算させる
        // 数字は文字コードとしての値となるため、補正が必要。
        const char zero = '0';
        foreach (var ch in line) {
            num += ch - zero;
        }

        // 出力する
        System.Console.WriteLine(num);
    }
}
